src/components/map/components/baselayer/index.js

```diff
@@ -94,38 +96,41 @@ export class VlMapBaseLayer extends vlElement(HTMLElement) {
     return this._createdVectorSource;
   }
 
-  _configureMap() {
+  async _configureMap() {
     if (this._map) {
       this._map.addBaseLayerAndOverlayMapLayer(
         this._createBaseLayer(),
         this._createBaseLayer()
       );
+      this._map.addBaseLayerAndOverlayMapLayer(
+        await this._createBaseLayer(),
+        await this._createBaseLayer()
+      );
+      // nodig anders is map initially blanc
+      this._map.render();
     }
   }
 
-  _createWMTSSource() {
-    const size = OlExtent.getWidth(this._projection.getExtent()) / 256;
-    const resolutions = new Array(16);
-    const matrixIds = new Array(16);
-    for (let z = 0; z < 16; ++z) {
-      resolutions[z] = size / Math.pow(2, z);
-      matrixIds[z] = z;
-    }
+  async _createWMTSSource() {
+    const params = {
+      request: "getcapabilities",
+      service: "wmts",
+      version: "1.0.0",
+    };
 
-    return new OlWMTSSource({
-      url: this.url,
+    const urlConstruct = new URL(this.url);
+    urlConstruct.search = new URLSearchParams(params);
+
+    const response = await fetch(urlConstruct.href).then((response) =>
+      response.text()
+    );
+    const parser = new WMTSCapabilities();
+    const val = optionsFromCapabilities(parser.read(response), {
       layer: this.layer,
       matrixSet: "BPL72VL",
-      format: "image/png",
-      projection: this._projection,
-      tileGrid: new OlWMTSTileGrid({
-        extent: this._projection.getExtent(),
-        origin: OlExtent.getTopLeft(this._projection.getExtent()),
-        resolutions: resolutions,
-        matrixIds: matrixIds,
-      }),
-      style: "",
```

Comment:
We zijn zeker dat dit geen impact heeft op bestaande use cases?

Reply:
Ik vermoed dat dit functioneel dus "maak uw kaartlaag op basis van de capabilities ipv op basis van hardcoded values" is? Als dat er nog hetzelfde uitziet en hetzelfde doet is dat inderdaad goed, anders moeten we hier, gezien er al code van afhankelijk is, dit met een parameter regelen denk ik, welk van beide opties ge wilt.

Reply:
Ook af te checken met @SBylemans lijkt me.